src/main/java/com/weaverplatform/postgresql/database/Postgres.java

```
@@ -104,6 +114,57 @@ public Connection getConnection(String database) throws SQLException {
     return connectionPool.get(database).getConnection();
   }
 
+  public Connection getReusableConnection(String database, String code, boolean createNew) throws SQLException {
+    log.info("Get reusable connection for code: " + code);
+    if(!reusableConnections.containsKey(code)) {
+      if(createNew) {
+        Connection connection = getConnection(database);
+        reusableConnections.put(code, connection);
+      } else {
+        throw new WeaverError(RESULTSET_ERROR, "No open connection for code: " + code);
+      }
+    }
+    reusableConnectionsTimestamps.put(code, new Date().getTime());
+    return reusableConnections.get(code);
+  }
+
+  public boolean closeReusableConnection(String code) {
+    log.info("Close reusable connection for code: " + code);
+    if(reusableConnections.containsKey(code)) {
+      Connection connection = reusableConnections.remove(code);
+      reusableConnectionsTimestamps.remove(code);
+      openResultSets.remove(code);
+      try {
+        connection.close();
+        return true;
+      } catch (SQLException e) {
+        e.printStackTrace();
```

Use logging, not standard output streams, and this error should be passed up to the caller, not swallowed.

---

Done